components/textfield/metadata/textfield.yml

```diff
@@ -34,83 +77,257 @@ sections:
+        </div>
+      </div>
+
+      <div class="spectrum-Textfield spectrum-Textfield--sizeXL is-valid">
+        <label for="textfield-xl-valid" class="spectrum-FieldLabel spectrum-FieldLabel--sizeXL">Enter your name</label>
+        <div class="spectrum-Textfield-input-wrapper">
+          <svg class="spectrum-Icon spectrum-UIIcon-Checkmark300 spectrum-Textfield-validationIcon" focusable="false" aria-hidden="true">
+            <use xlink:href="#spectrum-css-icon-Checkmark300" />
+          </svg>
+          <input id="textfield-xl-valid" type="text" placeholder="Lisa Wilson" name="field" value="A valid input" class="spectrum-Textfield-input" pattern="[\w\s]+" required>
+        </div>
       </div>
 
   - id: textfield
     name: Valid (disabled)
     markup: |
       <div class="spectrum-Textfield is-valid is-disabled">
+      <label for="textfield-valid-disabled" class="spectrum-FieldLabel spectrum-FieldLabel--sizeM">Enter your name</label>
+      <div class="spectrum-Textfield-input-wrapper">
         <svg class="spectrum-Icon spectrum-UIIcon-Checkmark100 spectrum-Textfield-validationIcon" focusable="false" aria-hidden="true">
           <use xlink:href="#spectrum-css-icon-Checkmark100" />
         </svg>
-        <input type="text" placeholder="Enter your name" name="field" value="A valid input" class="spectrum-Textfield-input" pattern="[\w\s]+" required disabled>
+          <input id="textfield-valid-disabled" type="text" placeholder="Lisa Wilson" name="field" value="A valid input" class="spectrum-Textfield-input" pattern="[\w\s]+" required disabled>
+        </div>
       </div>
 
   - id: textfield
     name: Invalid
     description: |
       *Spectrum for Adobe Express uses a different icon. Use the `SX_Alert_18_N.svg` icon in the Express workflow icon set.*    
     markup: |
-      <div class="spectrum-Textfield is-invalid">
-        <svg class="spectrum-Icon spectrum-Icon--sizeM spectrum-Textfield-validationIcon" focusable="false" aria-hidden="true">
-          <use xlink:href="#spectrum-icon-18-Alert" />
-        </svg>
-        <input type="text" placeholder="Enter your name" name="field" value="Invalid input" class="spectrum-Textfield-input" pattern="[\d]+" required>
+      <div class="spectrum-Textfield spectrum-Textfield--sizeS is-invalid">
+        <label for="textfield-sm-invalid" class="spectrum-FieldLabel spectrum-FieldLabel--sizeS">Enter your name</label>
+        <div class="spectrum-Textfield-input-wrapper">
+          <svg class="spectrum-Icon spectrum-Icon--sizeS spectrum-Textfield-validationIcon" focusable="false" aria-hidden="true">
+            <use xlink:href="#spectrum-icon-18-Alert" />
+          </svg>
+          <input id="textfield-sm-invalid" type="text" placeholder="Lisa Wilson" name="field" value="Invalid input" class="spectrum-Textfield-input" pattern="[\d]+" required>
```

Comment:
Minor: make the input patterns consistent with the help texts e.g. `pattern=".{8,}"` for at "least 8 characters"

Reply:
Good call, updated in both textfield and textarea.